core/includes/image.inc

```diff
@@ -452,5 +451,91 @@ function image_save(stdClass $image, $destination = NULL) {
 }
 
 /**
+ * Determines the correct color index value for the background.
+ *
+ * @param $background
+ *   A string specifying an RGBA color in the formats: '#RGB', 'RGB', '#RGBA',
+ *   'RGBA', '#RRGGBB', 'RRGGBB', '#RRGGBBAA', 'RRGGBBAA' or '0xRRGGBBAA'.
+ * @param $image
+ *   An image object. The $image->resource, $image->info['width'], and
+ *   $image->info['height'] values will be modified by this call.
+ *
+ * @return $background
+ *   A color identifier or FALSE if the allocation failed. If no color is
+ *   provided this will default to transparent For images that support
+ *   transparency, otherwise it will be white.
+ *
+ * @see image_hex2rgba().
+ */
+function image_get_colorindex($background, stdClass &$image) {
```

Comment:
To match with coding standards, parameters that are by-reference should be first, followed by optional parameters. As `$background` can be optional, it should have a default of NULL and be the second parameter.

`$image` parameter also does not need the `&` by-reference indicator. All objects are passed by reference in PHP 5+.